R/peaks-functions.R

```diff
@@ -49,25 +49,24 @@ NULL
 #' @importClassesFrom IRanges IRanges
 #'
 #' @noRd
-.peaks_remove <- function(x, spectrumMsLevel, centroided = NA,
-                          threshold = "min", msLevel = spectrumMsLevel, ...) {
+.peaks_replace_intensity <- function(x, spectrumMsLevel, centroided = NA,
+                                     threshold = min, value = 0,
+                                     msLevel = spectrumMsLevel, ...) {
     if (!spectrumMsLevel %in% msLevel || !length(x))
         return(x)
     if (is.na(centroided)) {
         warning("Centroided undefined (NA): keeping spectrum as is.")
         return(x)
     }
-    if (threshold == "min")
-        threshold <- min(x[x[, "intensity"] > 0, "intensity"])
-    if (!is.numeric(threshold))
-        stop("'threshold' must either be 'min' or numeric.")
+    if (is.function(threshold))
+        threshold <- threshold(x[, "intensity"], na.rm = TRUE)
```

Comment:
Wouldn't be `threshold(x[, "intensity"], na.rm = TRUE, ...)` useful?
Maybe that would be to complicated. Just keep it for now.